src/Compilers/CSharp/Portable/Symbols/Source/SourceTypeParameterSymbol.cs

```diff
@@ -179,19 +178,24 @@ internal virtual CustomAttributesBag<CSharpAttributeData> GetAttributesBag(Diagn
                 {
                     lazyAttributesStored = LoadAndValidateAttributes(
                         OneOrMany.Create(this.MergedAttributeDeclarationSyntaxLists),
-                        ref _lazyCustomAttributesBag,
-                        addToDiagnostics: diagnosticsOpt);
+                        ref _lazyCustomAttributesBag);
                 }
                 else
                 {
                     var typeParameter = (SourceTypeParameterSymbolBase)sourceMethod.SourcePartialDefinition.TypeParameters[_ordinal];
-                    CustomAttributesBag<CSharpAttributeData> attributesBag = typeParameter.GetAttributesBag(diagnosticsOpt);
+                    CustomAttributesBag<CSharpAttributeData> attributesBag = typeParameter.GetAttributesBag();
 
                     lazyAttributesStored = Interlocked.CompareExchange(ref _lazyCustomAttributesBag, attributesBag, null) == null;
                 }
 
                 if (lazyAttributesStored)
                 {
+                    if (ContainingSymbol is LocalFunctionSymbol localFunc)
```

Comment:
>if [](start = 20, length = 2)

It feels uncomfortable to see the implementation of local functions leaking down to this shared helper. Since this could be a purely syntactic check, would it make sense to do it eagerly closer to local function's implementation type?